tests/config/read.c

```diff
@@ -928,3 +928,64 @@ void test_config_read__nosection(void)
 	git_buf_dispose(&buf);
 	git_config_free(cfg);
 }
+
+enum {
+	MAP_TRUE = 0,
+	MAP_FALSE = 1,
+	MAP_ALWAYS = 2
+};
+
+static git_configmap _test_map1[] = {
+	{GIT_CONFIGMAP_STRING, "always", MAP_ALWAYS},
+	{GIT_CONFIGMAP_FALSE, NULL, MAP_FALSE},
+	{GIT_CONFIGMAP_TRUE, NULL, MAP_TRUE},
+};
+
+static git_configmap _test_map2[] = {
+	{GIT_CONFIGMAP_INT32, NULL, 0},
+};
+
+void test_config_read__get_mapped(void)
+{
+	git_config *cfg;
+	int val;
+	int known_good;
+
+	cl_set_cleanup(&clean_test_config, NULL);
+	cl_git_mkfile("./testconfig", "[header]\n  key1 = 1\n  key2 = true\n  key3\n  key4 = always\n  key5 = false\n  key6 = 0\n  key7 = never\n");
```

Comment:
do you mind also adding `on` and `off`? those two values don't seem to be covered by any other test cases.

(optional: for clarity, would it make sense to split the individual lines in the file? that can be achieved similarly to

https://github.com/libgit2/libgit2/blob/1e987525b3bd86fc29e7ff67d973861efd22faf7/tests/config/include.c#L61-L64

it took me a while to find the key that did not have a value ^^;; )